src/components/touchspin/touchspin.js

```diff
@@ -59,46 +72,39 @@ class TouchSpin extends React.Component {
     clearTimeout(this.timer);
   };
 
-  onMin = () => {
-    const { maxValue, minValue, value } = this.state;
-    let updatedValue = value - 1 > minValue ? value - 1 : minValue;
+  onMinus = () => {
+    const { displayValue, minValue, maxValue } = this.state;
 
-    if (updatedValue > maxValue) {
-      updatedValue = maxValue;
-    }
-
-    this.onUpdateValue({ target: { value: updatedValue } });
+    this.setState({
+      displayValue: this.normalizeBetween(displayValue - 1, minValue, maxValue)
+    });
   };
 
-  onMax = () => {
-    const { maxValue, minValue, value } = this.state;
-    let updatedValue = value + 1 < maxValue ? value + 1 : maxValue;
+  onPlus = () => {
+    const { displayValue, minValue, maxValue } = this.state;
 
-    if (updatedValue < minValue) {
-      updatedValue = minValue;
-    }
-
-    this.onUpdateValue({ target: { value: updatedValue } });
+    this.setState({
+      displayValue: this.normalizeBetween(displayValue + 1, minValue, maxValue)
+    });
   };
 
-  onUpdateValue = event => {
-    const { maxValue, minValue } = this.state;
+  onUpdateValue = (_updatedValue, event) => {
     const { name, onChange } = this.props;
-    const { value } = event.target;
+    const { maxValue, minValue } = this.state;
 
-    let parsedValue = Number.parseInt(value, 10);
-    parsedValue = Number.isNaN(parsedValue) ? minValue : parsedValue;
```

Comment:
required:
going back through, this NaN check needs to be restored too, and possibly the rest of lines `84` through `90`

reason:
- Input fields regardless of whether they're declared "number" still return a string. the joys of HTML/JS interaction.
- this also covers the scenario when someone enters nothing but word, one of the reasons we call `parseInt`